### ISSUE TITLE
API to detect if 'Developer Options' are enabled in Android

### ISSUE THREAD

**Issue by Yandamuri** [react-native-community/discussions-and-proposals#746]
Now a days many end users(Non Developers) are enabling `Developer Options` in their `Android `mobiles. Mostly people are enabling  `Developer Options` to set the mock location app. Though setting `Mock location` option was provided for developer for testing purpose, Many end users are using it extensively  in a wrong way like sales or marketing people are mocking their location without going to that particular location. Preventing such a things in app is a big challenge for developer.

My proposal here is If `react-native` can provide an API to detect if `Developer Options` are enabled, it would be useful for developer to prevent things like using mock location apps.

**Comment by liamjones** [comment]
This already exists in community modules: https://github.com/GantMan/jail-monkey

**Comment by Yandamuri** [comment]
@liamjones This library doesn't seem like actively maintained. Many unresolved issues were there.

**Comment by ganeshmohane** [comment]
> @liamjones This library doesn't seem like actively maintained. Many unresolved issues were there.

did you got any other alternative?
i am also looking for it

**Comment by liamjones** [comment]
@ganeshmohane Nope, we're still using JailMonkey for now. I haven't had a chance to investigate alternatives yet (and no breakage/upgrade has pushed me into it yet either)

**Comment by ganeshmohane** [comment]
> @ganeshmohane Nope, we're still using JailMonkey for now. I haven't had a chance to investigate alternatives yet (and no breakage/upgrade has pushed me into it yet either)

ohk, I will also then go with JailMonkey, but if you found any better alternative then let me know
and Thanks for the reply

**Comment by liamjones** [comment]
Oh, also @ganeshmohane, if you only care about the "are developer settings enabled" bit of jail-monkey, [it's a very short piece of code](https://github.com/GantMan/jail-monkey/blob/master/android/src/main/java/com/gantix/JailMonkey/JailMonkeyModule.java#L45). You might want to just setup a native module yourself for this instead, there are guides for this on the RN website.

**Comment by ganeshmohane** [comment]
Ohk, 
Also As I am using expo JailMonkey is not an option for   me so then after some searching i found that expo-location lib gives option to check whether location is mocked or not, so i implemented that. 

![Screenshot_20241204-163240_ETimeTracker](https://github.com/user-attachments/assets/1966839f-e056-4178-b510-bc94955a8058)

Tested multiple fake location apps, working fine
